src/media/UMediaCore_FFmpeg.pas

```diff
@@ -568,54 +574,53 @@ function TPacketQueue.Put(Packet : PAVPacket): integer;
  *)
 function TPacketQueue.PutStatus(StatusFlag: integer; StatusInfo: Pointer): integer;
 var
-  TempPacket: PAVPacket;
+  Packet: PAVPacket;
 begin
-  // create temp. package
-  TempPacket := av_malloc(SizeOf(TAVPacket));
-  if (TempPacket = nil) then
+  Packet := av_packet_alloc();
+  if (Packet = nil) then
   begin
     Result := -1;
     Exit;
   end;
-  // init package
-  av_init_packet(TempPacket^);
-  TempPacket^.data  := Pointer(STATUS_PACKET);
-  TempPacket^.flags := StatusFlag;
+
+  Packet^.data  := Pointer(STATUS_PACKET);
+  Packet^.flags := StatusFlag;
   {$IF LIBAVCODEC_VERSION_MAJOR >= 59}
-  TempPacket^.opaque  := StatusInfo;
+  Packet^.opaque  := StatusInfo;
   {$ELSE}
-  TempPacket^.side_data := StatusInfo;
+  Packet^.side_data := StatusInfo;
   {$ENDIF}
-  // put a copy of the package into the queue
-  Result := Put(TempPacket);
-  // data has been copied -> delete temp. package
-  av_free(TempPacket);
+
+  Result := Put(Packet);
 end;
 
-procedure TPacketQueue.FreeStatusInfo(var Packet: TAVPacket);
+procedure TPacketQueue.FreeStatusInfo(Packet: PAVPacket);
 begin
   {$IF LIBAVCODEC_VERSION_MAJOR >= 59}
-  if (Packet.opaque <> nil) then
-    FreeMem(Packet.opaque);
+  if (Packet^.opaque <> nil) then
+  begin
```

Comment:
Why did you add begin/else here?

And we should remove this and the following nil check eventually. It is allowed to pass nil to FreeMemAndNil/FreeMem. But if you don't want to do it here, I'll make a PR afterwards.